plugins/pulp_rpm/plugins/distributors/yum/publish.py

```diff
@@ -625,6 +634,58 @@ def finalize(self):
                 add_metadata_file_metadata('updateinfo', self.context.metadata_file_path,
                                            self.context.checksum)
 
+    def _get_pkglist_to_publish(self, erratum_unit):
+        """
+        Make a list of packages which will be listed in the published erratum.
+
+        It is possible to filter packages only if `repo_unit_nevra` is available,
+        otherwise publish everything except duplicated collections.
+
+        :param erratum_unit: the erratum unit to analyze
+        :param type: pulp_rpm.plugins.db.models.Errata
+        :return: list of pkglists to publish
+        :rtype: list of dicts
+        """
+        pkglists_to_publish = []
+
+        # If we can pull a repo_id off the conduit, use that to generate repo-specific nevra
+        if self.context.conduit and hasattr(self.context.conduit, 'repo_id'):
+            repo_unit_nevra = self.context.get_repo_unit_nevra(erratum_unit)
+        else:
+            repo_unit_nevra = None
+
+        seen_pkglists = set()
+        for pkglist in erratum_unit.pkglist:
+            packages = tuple(sorted(p['filename'] for p in pkglist['packages']))
+            if packages in seen_pkglists:
+                continue
+            seen_pkglists.add(packages)
+
+            if repo_unit_nevra is None:
+                pkglists_to_publish.append(pkglist)
+                continue
+
+            packages_to_publish = []
+            for package in pkglist['packages']:
+                package_nevra = {'name': package['name'],
+                                 'version': package['version'],
+                                 'release': package['release'],
+                                 'epoch': package['epoch'] or '0',
+                                 'arch': package['arch']}
+
+                if package_nevra not in repo_unit_nevra:
+                    # current package not in the specified repo, don't add it to the output
```

Comment:
This is a nit, but when the logic moved from the xml writer to here, this code is no longer concerned with the outputting of packages to XML. `s/output/pkglist/`.